infra/bridge/src/controllers/promotions.js

```diff
@@ -0,0 +1,119 @@
+'use strict'
+
+const express = require('express')
+const router = express.Router()
+const { getAsync } = require('../utils/redis')
+
+const Attestation = require('./../models/attestation')
+
+// TODO: Add request body validation
+router.post('/verify', async (req, res) => {
+  const { type, socialNetwork, identity } = req.body
+
+  const attestation = await Attestation.findOne({
+    ethAddress: identity,
+    method: socialNetwork
+  })
+
+  if (!attestation) {
+    return res.status(400).send({
+      success: false,
+      errors: [`Attestation missing`]
+    })
+  }
+
+  const redisKey = `${socialNetwork.toLowerCase()}/${type.toLowerCase()}/${
+    attestation.value
+  }`
+  let tries = 0
+  const interval = setInterval(async () => {
```

Comment:
@franckc Can probably wrap this whole `setInterval` thing into a `Promise` and `await` it. But not sure if that is the right way. Any thoughts on this?

Reply:
One alternative that might be simpler to than setInterval is to define a tick() function like [this](https://github.com/OriginProtocol/origin/blob/master/infra/relayer/src/purse.js#L59).

Then have a loop like
do {
  ...
} while (retries < 300 && await tick())